.github/workflows/test.yml

```diff
@@ -0,0 +1,227 @@
+name: test
+
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      rose_ref:
+        description: The Rose branch to test against
+        required: true
+      fcm_ref:
+        description: The FCM branch to test against
+        required: false
+      fcm_repo:
+        description: The FCM repo to test against
+        required: false
+      cylc_ref:
+        description: The Cylc branch to test against
+        required: false
+      cylc_repo:
+        description: The Cylc repo to test against
+        required: false
+
+defaults:
+  run:
+    shell: bash  # macos default shell is zsh
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 45
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest']
+        python-version: ['3.7']
```

Comment:
Not 3.8 & 3.9 yet?

Reply:
Was just keeping the test matrix to a minimum whilst I went through the CI test failure loop.